packages/fast-components/src/dialog/dialog.ts

```diff
@@ -0,0 +1,145 @@
+import { attr, FastElement } from "@microsoft/fast-element";
+import { keyCodeEscape, keyCodeTab } from "@microsoft/fast-web-utilities";
+import tabbable from "tabbable";
+
+export class Dialog extends FastElement {
+    @attr
+    public modal: boolean = true;
+
+    @attr
+    public hidden: boolean = false;
+
+    @attr({ attribute: "trap-focus" })
+    public trapFocus: boolean = true;
+
+    @attr({ attribute: "aria-describedby" })
+    public ariaDescribedby: string;
+
+    @attr({ attribute: "aria-labelledby" })
+    public ariaLabelledby: string;
+
+    @attr({ attribute: "aria-label" })
+    public ariaLabel: string;
+
+    public dialog: HTMLDivElement;
+
+    public dismiss(): void {
+        this.dispatchEvent(
+            new CustomEvent("dismiss", {
+                bubbles: true,
+                composed: true,
+            })
+        );
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        document.addEventListener("keydown", this.handleDocumentKeydown);
+
+        if (this.shouldDialogTrapFocus()) {
```

Comment:
Adding and removing handlers should probably be a reaction to the `trapFocus` property. With this implementation, if `trapFocus` is changed during the component then lifecycle handlers may not be attached when they should be, or may be attached when they aren't needed.

You would still want to clean up in disconnectedCallback if handlers are attached

Reply:
I took a shot at this if. Resolving this only as it's outdated now.